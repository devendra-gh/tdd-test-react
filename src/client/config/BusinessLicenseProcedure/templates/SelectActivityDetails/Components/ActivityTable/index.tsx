import React from 'react';
import { IVariables } from '@tamm/app-composer';
import Table from '@tamm/ui-lib-v2-table';
import RadioGroup from '@tamm/ui-lib-v2-radio-group';
import { noRefCheck } from 'client/config/BusinessLicenseProcedure/utils';

function ActivityTable(props: IVariables) {
  const { i18n, formSelectActivity, locale } = props;
  const { activities, activity, showTable } = formSelectActivity;

  const onChange = (type: string, value: string) =>
    props.onChange(type, value, props);

  const getRadio = (activityCode: string) => {
    const businessActivity =
      activities &&
      activities.find((x: IVariables) => x.activityCode === activityCode);
    return (
      businessActivity && (
        <RadioGroup.Radio
          checked={businessActivity.activityCode === activity}
          onChange={() => onChange('activity', businessActivity.activityCode)}
          label={
            locale === 'en'
              ? businessActivity.activity
              : businessActivity.activityAr
          }
        />
      )
    );
  };

  return (
    <>
      {showTable && (
        <div className="table-container">
          <Table
            i18n={i18n}
            columns={[
              {
                id: 'checkbox',
                title: i18n('table.header.activityName'),
                render: (a: IVariables, activityCode: string, c: IVariables) =>
                  getRadio(activityCode),
              },
              {
                align: 'end',
                id: 'no',
                title: i18n('table.header.activityNumber'),
              },
            ]}
            items={
              activities &&
              activities.map((businessActivity: IVariables) => ({
                id: businessActivity.activityCode,
                no: businessActivity.activityCode,
              }))
            }
            onClick={noRefCheck}
            size="default"
            title={i18n('table.header.listOfActivities')}
          />
        </div>
      )}
    </>
  );
}

export default ActivityTable;
