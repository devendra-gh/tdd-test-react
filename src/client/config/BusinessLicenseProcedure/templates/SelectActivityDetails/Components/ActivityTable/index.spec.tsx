import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router';
import { IVariables } from '@tamm/app-composer';
import { Viewport, ViewportProvider } from '@tamm/ui-lib-v2-viewport';
import ActivityTable from './index';

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('FormTemplate', () => {
  let props: IVariables;

  beforeEach(() => {
    props = {
      history: {
        push: jest.fn(),
      },
      actions: {
        stepsStatus: {
          update: jest.fn(),
        },
        formSelectActivity: {
          update: jest.fn(),
        },
      },
      getInitialState: jest.fn(),
      formSelectActivity: {
        activities: [
          {
            activityCode: 'activityCode',
            activity: 'businessActivity.activity',
            activityAr: 'businessActivity.activityAr',
          },
        ],
        showTable: true,
        category: 'category',
        subCategory: 'subCategory',
        inputType: 'CATEGORY',
        loading: true,
      },
      onChange: jest.fn(),
      onSubmit: jest.fn(),
      i18n: (key: string) => key,
      title: 'FormTemplate',
      subTitle: 'subTitle',
      description: 'desc',
      locale: 'en',
      currentStep: 'economic_licence',
      currentSubStep: 'payment',
      buttons: [
        {
          label: 'btn',
          onClick: jest.fn(),
        },
      ],
      steps: [],
      stepStatus: {
        economic_name: 'economic_name',
        'economic_name.payment': 'economic_name.payment',
      },
      list: {
        labelHeading: 'label',
        valueHeading: 'value',
        listDetails: {
          label: 'label',
          value: 'value',
        },
      },
      totalSection: 1000,
    };
  });

  it('should render listOfActivities when showTable true - english', async () => {
    const { getByText, container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <ActivityTable {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(getByText('table.header.listOfActivities')).toBeInTheDocument();
    expect(getByText('businessActivity.activity')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render listOfActivities when showTable true - arabic', async () => {
    props.locale = 'ar';
    const { getByText, container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <ActivityTable {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(getByText('table.header.listOfActivities')).toBeInTheDocument();
    expect(getByText('businessActivity.activityAr')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call onchange when radio is clicked', async () => {
    const { container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <ActivityTable {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    fireEvent.click(container.querySelectorAll('.ui-lib-radio__input')[0]);
    expect(props.onChange).toBeCalled();
  });
});
