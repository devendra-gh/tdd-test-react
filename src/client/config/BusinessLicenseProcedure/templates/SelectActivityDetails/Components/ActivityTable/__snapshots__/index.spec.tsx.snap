// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTemplate should render listOfActivities when showTable true - arabic 1`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <div
    class="table-container"
  >
    <div
      class="ui-lib-table"
    >
      <div
        class="ui-lib-table__pre"
      >
        <div
          class="ui-lib-table__pre-title ui-lib-heading_h4"
        >
          table.header.listOfActivities
        </div>
      </div>
      <div
        class="ui-lib-table__root"
        data-round-bottom="true"
        data-round-top="false"
      >
        <table
          class="ui-lib-table__root-grid"
        >
          <thead
            class="ui-lib-table-header"
          >
            <tr
              class="ui-lib-table-row"
              data-highlighted="false"
              data-ui-type="default"
              data-underlined="false"
              style="vertical-align: top;"
            >
              <td
                aria-label="checkbox"
                class="ui-lib-table-cell"
                data-align="start"
                data-collapsed="false"
                data-disabled="false"
                data-fit-content-width="false"
                data-hoverable="false"
                data-is-extra="false"
                data-is-header="true"
                data-size="default"
                data-sortable="false"
                data-ui-type="header"
                role="gridcell"
                tabindex="0"
              >
                <span
                  class="ui-lib-table-cell_bdi"
                >
                  table.header.activityName
                </span>
                
              </td>
              <td
                aria-label="no"
                class="ui-lib-table-cell"
                data-align="end"
                data-collapsed="false"
                data-disabled="false"
                data-fit-content-width="false"
                data-hoverable="false"
                data-is-extra="false"
                data-is-header="true"
                data-size="default"
                data-sortable="false"
                data-ui-type="header"
                role="gridcell"
                tabindex="0"
              >
                <span
                  class="ui-lib-table-cell_bdi"
                >
                  table.header.activityNumber
                </span>
                
              </td>
            </tr>
          </thead>
          <tbody>
            <tr
              class="ui-lib-table-row"
              data-highlighted="false"
              data-ui-type="regular"
              data-underlined="false"
              style="vertical-align: top;"
            >
              <td
                aria-label="checkbox"
                class="ui-lib-table-cell"
                data-align="start"
                data-collapsed="false"
                data-disabled="false"
                data-fit-content-width="false"
                data-hoverable="false"
                data-is-extra="false"
                data-is-header="false"
                data-size="default"
                data-sortable="false"
                data-ui-type="regular"
                role="gridcell"
                tabindex="0"
              >
                <label
                  class="ui-lib-radio ui-lib-radio_default"
                  for="uil_radio_button.2"
                >
                  <input
                    class="ui-lib-radio__input"
                    data-index="0"
                    id="uil_radio_button.2"
                    name=""
                    type="radio"
                    value=""
                  />
                  <div
                    class="ui-lib-radio__box"
                  >
                    <div
                      class="ui-lib-radio__box-dot"
                    />
                  </div>
                  <div
                    class="ui-lib-radio-label"
                  >
                    businessActivity.activityAr
                  </div>
                </label>
                
              </td>
              <td
                aria-label="no"
                class="ui-lib-table-cell"
                data-align="end"
                data-collapsed="false"
                data-disabled="false"
                data-fit-content-width="false"
                data-hoverable="false"
                data-is-extra="false"
                data-is-header="false"
                data-size="default"
                data-sortable="false"
                data-ui-type="regular"
                role="gridcell"
                tabindex="0"
              >
                <span
                  class="ui-lib-table-cell_bdi"
                >
                  activityCode
                </span>
                
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`FormTemplate should render listOfActivities when showTable true - english 1`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <div
    class="table-container"
  >
    <div
      class="ui-lib-table"
    >
      <div
        class="ui-lib-table__pre"
      >
        <div
          class="ui-lib-table__pre-title ui-lib-heading_h4"
        >
          table.header.listOfActivities
        </div>
      </div>
      <div
        class="ui-lib-table__root"
        data-round-bottom="true"
        data-round-top="false"
      >
        <table
          class="ui-lib-table__root-grid"
        >
          <thead
            class="ui-lib-table-header"
          >
            <tr
              class="ui-lib-table-row"
              data-highlighted="false"
              data-ui-type="default"
              data-underlined="false"
              style="vertical-align: top;"
            >
              <td
                aria-label="checkbox"
                class="ui-lib-table-cell"
                data-align="start"
                data-collapsed="false"
                data-disabled="false"
                data-fit-content-width="false"
                data-hoverable="false"
                data-is-extra="false"
                data-is-header="true"
                data-size="default"
                data-sortable="false"
                data-ui-type="header"
                role="gridcell"
                tabindex="0"
              >
                <span
                  class="ui-lib-table-cell_bdi"
                >
                  table.header.activityName
                </span>
                
              </td>
              <td
                aria-label="no"
                class="ui-lib-table-cell"
                data-align="end"
                data-collapsed="false"
                data-disabled="false"
                data-fit-content-width="false"
                data-hoverable="false"
                data-is-extra="false"
                data-is-header="true"
                data-size="default"
                data-sortable="false"
                data-ui-type="header"
                role="gridcell"
                tabindex="0"
              >
                <span
                  class="ui-lib-table-cell_bdi"
                >
                  table.header.activityNumber
                </span>
                
              </td>
            </tr>
          </thead>
          <tbody>
            <tr
              class="ui-lib-table-row"
              data-highlighted="false"
              data-ui-type="regular"
              data-underlined="false"
              style="vertical-align: top;"
            >
              <td
                aria-label="checkbox"
                class="ui-lib-table-cell"
                data-align="start"
                data-collapsed="false"
                data-disabled="false"
                data-fit-content-width="false"
                data-hoverable="false"
                data-is-extra="false"
                data-is-header="false"
                data-size="default"
                data-sortable="false"
                data-ui-type="regular"
                role="gridcell"
                tabindex="0"
              >
                <label
                  class="ui-lib-radio ui-lib-radio_default"
                  for="uil_radio_button.1"
                >
                  <input
                    class="ui-lib-radio__input"
                    data-index="0"
                    id="uil_radio_button.1"
                    name=""
                    type="radio"
                    value=""
                  />
                  <div
                    class="ui-lib-radio__box"
                  >
                    <div
                      class="ui-lib-radio__box-dot"
                    />
                  </div>
                  <div
                    class="ui-lib-radio-label"
                  >
                    businessActivity.activity
                  </div>
                </label>
                
              </td>
              <td
                aria-label="no"
                class="ui-lib-table-cell"
                data-align="end"
                data-collapsed="false"
                data-disabled="false"
                data-fit-content-width="false"
                data-hoverable="false"
                data-is-extra="false"
                data-is-header="false"
                data-size="default"
                data-sortable="false"
                data-ui-type="regular"
                role="gridcell"
                tabindex="0"
              >
                <span
                  class="ui-lib-table-cell_bdi"
                >
                  activityCode
                </span>
                
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      
    </div>
  </div>
</div>
`;
